plasmapy/analysis/swept_langmuir/floating_potential.py

```diff
@@ -0,0 +1,303 @@
+        (e.g. the
+        `~plasmapy.analysis.fit_functions.ExponentialPlusOffset.rsq`
+        property of
+        `~plasmapy.analysis.fit_functions.ExponentialPlusOffset`).
+
+    func: sub-class of `~plasmapy.analysis.fit_functions.AbstractFitFunction`
+        The callable function :math:`f(x)` repressing the fit and its results.
+
+    islands: `List[slice]`
+        List of `slice` objects representing the indices of the identified
+        crossing-islands.
+
+    indices: `slice`
+        A `slice` object representing the indices of ``voltage`` and ``current``
+        arrays used for the fit.
+
+    """
+    rtn = FloatingPotentialResults(
+        vf=np.nan,
+        vf_err=np.nan,
+        rsq=None,
+        func=None,
+        islands=None,
+        indices=None
+    )._asdict()
+
+    fit_funcs = {
+        "linear": {
+            "func": ffuncs.Linear(),
+            "min_point_factor": 0.1,
+        },
+        "exponential": {
+            "func": ffuncs.ExponentialPlusOffset(),
+            "min_point_factor": 0.2,
+        },
+    }
+    try:
+        min_point_factor = fit_funcs[fit_type]["min_point_factor"]
+        fit_func = fit_funcs[fit_type]["func"]
+        rtn["func"] = fit_func
+    except KeyError:
+        raise KeyError(
+            f"Requested fit function '{fit_type}' is  not a valid option.  "
+            f"Examine kwarg 'fit_curve' for valid options."
+        )
+
+    if current.min() > 0.0 or current.max() < 0:
+        warn("The Langmuir sweep has no floating potential.")
+
+        return FloatingPotentialResults(**rtn)
+
+    # check voltage is monotonically increasing/decreasing
+    voltage_diff = np.diff(voltage)
+    if not (np.all(voltage_diff >= 0) or np.all(voltage_diff <= 0)):
+        warn("The voltage array is not monotonically increasing or decreasing.")
+
+        return FloatingPotentialResults(**rtn)
+
+    # condition kwarg threshold
+    if isinstance(threshold, (int, float)):
+        threshold = int(np.round(threshold))
+        if threshold < 1:
+            warn(f"threshold ({threshold}) is less than 1 and needs to be"
+                 f" an int >= 1, using a value of 1")
+            threshold = 1
+    else:
+        warn(f"threshold is NOT a integer >= 1, using a value of 1")
+
+    # condition min_points
+    if min_points is None:
+        min_points = int(np.max([5, np.around(min_point_factor * voltage.size)]))
+    elif min_points == 0:
+        # this signals to use all points
+        pass
+    elif 0 < min_points < 1:
+        min_points = int(np.round(min_points * voltage.size))
+    elif min_points >= 1:
+        min_points = int(np.round(min_points))
+    else:
+        raise ValueError(f"Got {min_points}, but 'min_points' must be an int or float "
+                         f"greater than or equal to 0.")
+
+    # find possible crossing points (cp)
+    lower_vals = np.where(current <= 0, True, False)
+    upper_vals = np.where(0 <= current, True, False)
```

Comment:
Aren't these just

```suggestion
    lower_vals = current <= 0
    upper_vals = 0 <= current
```

Reply:
Yeah, it's the same thing.  I've just always done it with `np.where` because that's how I first learned it.

Reply:
But my way is significantly slower, so I'll change it up...

<img width="607" alt="image" src="https://user-images.githubusercontent.com/29869348/94302946-ea696200-ff21-11ea-8d82-a2e1a4e242c8.png">

Reply:
It's unlikely that anything equals zero, but should you use, for example ,
lower_vals = current <= 0
upper_vals = 0 < current

Reply:
That's a fringe condition.  The current implementation could accidentally collect points around `current == 0`.  Taking your suggestion in mind, I think this would be the right approach...

```python
lower_vals = current < 0
upper_vals = 0 < current
cp_exact = (current == 0).nonzero()[0]
```